config/config.go

```diff
@@ -224,6 +225,10 @@ type Config struct {
 	StorageConfig     StorageConfig   `yaml:"storage,omitempty"`
 	TracingConfig     TracingConfig   `yaml:"tracing,omitempty"`
 
+	// Secret management:
+	kubernetes.ClientConfig  `yaml:"kubernetes_sp_config,omitempty"`
```

Comment:
What's the reason behind this VS inlining those configs exactly where we reference those? We can deduplicate and manage it on our side, but forcing user to configure same information twice might be not necessary 🤔 

EDIT: Ok I think we do this to avoid provider deps in common.... https://github.com/TheSpiritXIII/prometheus-common/blob/v0.45.0-gmp.1/config/http_config.go#L138

This is good enough for MVP, but I would double check alternatives -- generally  making configuration surface worse for user to avoid code duplication is not a right move. I wonder what we can do long term.  🤔 

Other can moving secretful HTTP client object back to Prometheus, I don't have a better idea.. 

The counter argument for references is the fact similar thing is in Kubernetes volume mounting.. 🤔 Just totally new pattern in Prometheus configuration 🤔

Reply:
Hm, maybe I misunderstood at the end, I don't see you proposing "ref" vars here: https://github.com/prometheus/common/pull/538/files (maybe you planned to do it in another PR)

Reply:
Note, the `kubernetes_sp_config` would definitely change in the final upstream revision. Users may want different Kubernetes configurations per-secret and we would need to expose that.